src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -894,6 +894,25 @@ public T WithB2CAuthority(string authorityUri)
             return (T)this;
         }
 
+        public T WithGenericAuthority(string authorityUri, string discoveryDocumentAddress = null)
+        {
+            var authorityInfo = AuthorityInfo.FromGenericAuthority(authorityUri, discoveryDocumentAddress);
+            Config.Authority = GenericAuthority.CreateAuthority(authorityInfo);
+            return (T)this;
+        }
+        
+        public T WithDocumentRetriever(IDocumentRetriever documentRetriever)
```

Comment:
Are you missing some files? This `IDocumentRetriever` seems to be undefined.

Reply:
`IDocumentRetriever` comes from Microsoft.IdentityModel.Protocols nuget package. I took a dependency on Microsoft.IdentityModel.Protocols.OpenIdConnect in Microsoft.IdentityModel.Client.